src/pages/Details.jsx

```diff
@@ -19,7 +19,7 @@ const Details = () => {
   }, [dispatch]);
 
   const countryIndex = countries.findIndex((country) => country.name.common === name );
-
+  console.log(countries[countryIndex].name.common);
   return (
```

Comment:
- [ ] As requested by the previous code reviewer, please kindly fix the errors with the tests, checkout this [article](https://reactjs.org/docs/error-boundaries.html), it should be of help.

Reply:
Hi @Mihndim2020 ,  Good resource you sent me, but I have implemented this earlier in many ways and still not finding my solution. Error boundaries work with class-based components.

Reply:
@Mihndim2020, and I am using functional-based components.

Reply:
@Davidosky007  @Mihndim2020 , I am still not able to fix it please, I wish I can do this later.  I am worn out.  from a blog, they said it is from the dynamic routing but i have not been able to figure out the solution.